mtp_api/apps/transaction/tests/test_cashbook_views.py

```diff
@@ -36,800 +34,584 @@ def _get_authorised_user(self):
+            'status': TRANSACTION_STATUS.AVAILABLE
+        })
+
+    def test_WITH_status_locked_DEFAULT_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.LOCKED
+        })
+
+    def test_WITH_status_credited_DEFAULT_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.CREDITED
+        })
+
+    def test_WITH_status_available_AND_prison_DEFAULT_user(self):
+        self._test_response_with_filters(filters={
+            'prison': self.prisons[0].pk
+        })
+
+    def test_WITH_status_locked_AND_prison_DEFAULT_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.LOCKED,
+            'prison': self.prisons[0].pk
+        })
+
+    def test_WITH_status_credited_AND_prison_DEFAULT_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.CREDITED,
+            'prison': self.prisons[0].pk
+        })
+
+    def test_WITH_status_available_AND_user_DEFAULT_prison(self):
+        self._test_response_with_filters(filters={
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_locked_AND_user_DEFAULT_prison(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.LOCKED,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_credited_AND_user_DEFAULT_prison(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.CREDITED,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_available_AND_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'prison': self.prisons[0].pk,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_locked_AND_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.LOCKED,
+            'prison': self.prisons[0].pk,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_status_credited_AND_prison_AND_user(self):
+        self._test_response_with_filters(filters={
+            'status': TRANSACTION_STATUS.CREDITED,
+            'prison': self.prisons[0].pk,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_prison_DEFAULT_status_user(self):
+        self._test_response_with_filters(filters={
+            'prison': self.prisons[0].pk
+        })
+
+    def test_WITH_prison_AND_user_DEFAULT_status(self):
+        self._test_response_with_filters(filters={
+            'prison': self.prisons[0].pk,
+            'user': self.prison_clerks[1].pk
+        })
+
+    def test_WITH_user_DEFAULT_status_prison(self):
+        self._test_response_with_filters(filters={
+            'user': self.prison_clerks[1].pk
+        })
```

Comment:
Sorry, I wasn't entirely clear: I'm not saying one class per function here. Something a bit like this (with details omitted for clarity:

```python
class TransactionListWithDefaultPrisonAndUserTestCase(…):
    def test_no_filter_returns_all_transactions(…):
        # …
    def test_filter_by_available_status_returns_only_available_transactions(…):
        # …
    def test_filter_by_credited_status_returns_only_credited_transactions(…):
        # …
```

… and so on.

That way, much of the initial setup logic in `_test_response_with_filters` can be moved into class-level `setUp` methods (maybe with a base class for all the tests or something), and each test method can directly make the request and test the responses.